xml/operating_system_windows.xml

```diff
@@ -0,0 +1,98 @@
+    <description>Windows 2000</description>
+    <example os.edition="Professional">Windows 2000 Professional</example>
+    <example os.edition="Advanced Server">Windows 2000 Advanced Server</example>
+    <param pos="0" name="os.vendor" value="Microsoft"/>
+    <param pos="0" name="os.family" value="Windows"/>
+    <param pos="0" name="os.product" value="Windows 2000"/>
+    <param pos="1" name="os.edition"/>
+    <param pos="2" name="os.version"/>
+  </fingerprint>
+  <fingerprint pattern="^(?i:(?:Microsoft )?Windows NT (\d.\d{1,2})?(?:\s)?([a-z]+|[a-z]+\s[a-z]+)?)$">
+    <description>Windows NT</description>
+    <example os.version="3.51" os.edition="Server">Windows NT 3.51 Server</example>
+    <example os.edition="Workstation">Windows NT Workstation</example>
+    <example os.version="4.0" os.edition="Workstation">Windows NT 4.0 Workstation</example>
+    <example os.edition="Advanced Server">Windows NT Advanced Server</example>
+    <param pos="0" name="os.vendor" value="Microsoft"/>
+    <param pos="0" name="os.family" value="Windows"/>
+    <param pos="0" name="os.product" value="NT"/>
+    <param pos="1" name="os.version"/>
+    <param pos="2" name="os.edition"/>
+  </fingerprint>
+  <fingerprint pattern="^(?i:(?:Microsoft )?Windows 10 Mobile(?:\s(?!Edition)([a-z]+))?(?: Edition)?)$">
+    <description>Windows 10 Mobile</description>
+    <example os.product="Windows 10 Mobile">Windows 10 Mobile Edition</example>
+    <example os.product="Windows 10 Mobile" os.edition="Enterprise">Windows 10 Mobile Enterprise Edition</example>
+    <param pos="0" name="os.vendor" value="Microsoft"/>
+    <param pos="0" name="os.family" value="Windows"/>
+    <param pos="0" name="os.product" value="Windows 10 Mobile"/>
+    <param pos="1" name="os.edition"/>
+    <param pos="0" name="os.device" value="Mobile"/>
+  </fingerprint>
+  <fingerprint pattern="^(?i:(?:Microsoft )?Windows Phone (\d|\d\.\d)?)$">
+    <description>Windows Phone 7 and later</description>
+    <example os.version="7.5">Windows Phone 7.5</example>
+    <example os.version="8">Windows Phone 8</example>
+    <param pos="0" name="os.vendor" value="Microsoft"/>
+    <param pos="0" name="os.family" value="Windows"/>
+    <param pos="0" name="os.product" value="Windows Phone"/>
+    <param pos="1" name="os.version"/>
+    <param pos="0" name="os.device" value="Mobile"/>
+  </fingerprint>
+  <fingerprint pattern="^(?i:(?:Microsoft )?(Windows\s?(?:95|98|98SE|98 SE|98 Second Edition|ME|Millenium Edition)))$">
+    <description>Windows 9x</description>
+    <example os.product="Windows 98 SE">Windows 98 SE</example>
+    <param pos="0" name="os.vendor" value="Microsoft"/>
+    <param pos="0" name="os.family" value="Windows"/>
+    <param pos="1" name="os.product"/>
+  </fingerprint>
+  <fingerprint pattern="^(?i:(?:Microsoft )?Windows(?:\s?\w+?)*?\s?(\d+?(?:\.\d+?)*?)?)$">
+    <description>Windows catch-all</description>
```

Comment:
Catch-alls are catch-22s.  On the one hand,  yeah, you'll at least get a Windows fingerprint, but what if "Windows 2020" comes out?  Then with this file you won't know that you are missing an explicit match for that OS.